java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -85,6 +111,35 @@ ObjectTransform buildDefaultTransform() {
             .build();
     }
 
+    /**
+     * @return metadata to be written to the transformed object
+     */
+    public Map<String, String> getObjectMeta(String importBucket, String sourceKey, ObjectTransform transform) {
+        return Map.of(
+            BulkMetaData.INSTANCE_ID.getMetaDataKey(), hostEnvironment.getInstanceId(),
+            BulkMetaData.VERSION.getMetaDataKey(), config.getConfigPropertyAsOptional(ProxyConfigProperty.BUNDLE_FILENAME).orElse("unknown"),
+            BulkMetaData.ORIGINAL_OBJECT_KEY.getMetaDataKey(), importBucket + "/" + sourceKey
+        );
+    }
+
+    /**
+     * @return true if the object has already been sanitized by this proxy instance
+     */
+    public boolean hasBeenSanitized(Map<String, String> objectMeta) {
```

Comment:
probably I'd name this as `isCreatedByProxy` rather than coupling with function... could not be sanitized if the column config is wrong.

Reply:
it also shouldn't have metadata if column config was wrong ...